mldsa/fips202/fips202.c

```diff
@@ -559,15 +575,30 @@ __contract__(
  *              - uint64_t *s: pointer to input/output Keccak state
  *              - unsigned int r: rate in bytes (e.g., 168 for SHAKE128)
  **************************************************/
-static void keccak_squeezeblocks(uint8_t *out, size_t nblocks, uint64_t s[25],
-                                 unsigned int r)
+static void keccak_squeezeblocks(uint8_t *out, size_t nblocks,
+                                 uint64_t s[MLD_KECCAK_LANES], unsigned int r)
+__contract__(
+  requires(r < sizeof(uint64_t) * MLD_KECCAK_LANES)
```

Comment:
It might be better here to specifically restrict r to the only 2 legal values that are allowed.
e.g
`requires(r == SHAKE128_RATE || r == SHAKE256_RATE)`